src/engine.rs

```diff
@@ -0,0 +1,105 @@
+
+    let input: serde_json::Value = serde_json::from_reader(
+        std::fs::File::open(&input_path)
+            .map_err(|e| anyhow!("Couldn't load input {:?}: {}", &input_path, e))?,
+    )
+    .map_err(|e| anyhow!("Couldn't load input {:?}: {}", &input_path, e))?;
+    let input = serde_json::to_vec(&input)?;
+
+    let input_stream = wasi_common::pipe::ReadPipe::new(std::io::Cursor::new(input));
+    let output_stream = wasi_common::pipe::WritePipe::new_in_memory();
+    let error_stream = wasi_common::pipe::WritePipe::new_in_memory();
+
+    let runtime: Duration;
+
+    {
+        let mut linker = Linker::new(&engine);
+        wasmtime_wasi::add_to_linker(&mut linker, |s| s)?;
+        let wasi = WasiCtxBuilder::new()
+            .stdin(Box::new(input_stream))
+            .stdout(Box::new(output_stream.clone()))
+            .stderr(Box::new(error_stream.clone()))
+            .inherit_args()?
+            .build();
+        let mut store = Store::new(&engine, wasi);
+
+        linker.module(&mut store, "", &module)?;
+
+        let start = Instant::now();
+
+        let instance = linker.instantiate(&mut store, &module)?;
+        let module_result = instance
+            .get_typed_func::<(), (), _>(&mut store, "_start")?
+            .call(&mut store, ());
+
+        runtime = start.elapsed();
+
+        match module_result {
+            Ok(_) => {}
+            Err(e) => {
+                eprintln!("Error:\n{}", e);
+            }
+        }
+    };
+
+    let logs = error_stream
+        .try_into_inner()
+        .expect("Error stream reference still exists")
+        .into_inner();
+    let logs =
+        std::str::from_utf8(&logs).map_err(|e| anyhow!("Couldn't print Script Logs: {}", e))?;
+
+    let output = output_stream
+        .try_into_inner()
+        .expect("Output stream reference still exists")
+        .into_inner();
+    let output: serde_json::Value = serde_json::from_slice(output.as_slice())
+        .map_err(|e| anyhow!("Couldn't decode Script Output: {}", e))?;
+
+    let function_run_result = FunctionRunResult::new(runtime, output, logs.to_string());
+
+    Ok(function_run_result)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::path::Path;
+
+    #[test]
+    fn test_runtime_under_threshold() {
+        let function_run_result = run(
+            Path::new("tests/benchmarks/hello_world.wasm").to_path_buf(),
+            Path::new("tests/benchmarks/hello_world.json").to_path_buf(),
+        )
+        .unwrap();
+
+        assert!(function_run_result.runtime <= Duration::from_millis(5));
+    }
+
+    #[test]
+    fn test_runtime_over_threshold() {
+        let function_run_result = run(
+            Path::new("tests/benchmarks/sleeps.wasm").to_path_buf(),
+            Path::new("tests/benchmarks/sleeps.json").to_path_buf(),
+        )
+        .unwrap();
+
+        assert!(function_run_result.runtime > Duration::from_millis(5));
+    }
+}
```

Comment:
Good idea adding unit tests! 

I think it would also be useful to have a test with a Wasm module that reads from the input, logs some stuff, and outputs something. That way we can verify that the returned `FunctionRunResult` is what we expect.

For example, the input could be something like `{ value: 1 }` and the function is expected to output `value + 1`. We could also verify that it logs something like `hello world`.